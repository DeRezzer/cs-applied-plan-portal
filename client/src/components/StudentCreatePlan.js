import React from "react";
import EditPlan from "./EditPlan";
import CourseContainer from "./CourseContainer";
import Navbar from "./Navbar";

export default class StudentCreatePlan extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      courses: [],
      totalCredits: 0
    };

    this.updateCourses = this.updateCourses.bind(this);
    this.removeCourse = this.removeCourse.bind(this);
  }

  updateCourses(newCourses) {
    this.setState({
      courses: newCourses
    });
  }

  removeCourse(course) {
    const newCourses = this.state.courses;
    for (let i = 0; i < newCourses.length; i++) {
      if (newCourses[i].code === course.code) {
        newCourses.splice(i, 1);
      }
    }
    this.setState({
      courses: newCourses
    });
  }

  render() {
    return (
      <div className="student-create-plan">
        <Navbar showSearch={false} searchContent={null}/>
        <EditPlan courses={this.state.courses} remove={this.removeCourse}/>
        <CourseContainer updateCourses={this.updateCourses}/>
      </div>
    );
  }
}