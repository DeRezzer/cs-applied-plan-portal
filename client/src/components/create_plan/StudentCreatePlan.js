/** @jsx jsx */

import {useState, useEffect} from "react";
import EditPlan from "./EditPlan";
import CourseSearch from "./CourseSearch";
import Navbar from "../navbar/Navbar";
import PageSpinner from "../general/PageSpinner";
import {useParams} from "react-router-dom";
import PageInternalError from "../general/PageInternalError";
import PageNotFound from "../general/PageNotFound";
import {css, jsx} from "@emotion/core";

export default function StudentCreatePlan() {

  const [loading, setLoading] = useState(false);
  const [subloading, setSubloading] = useState(false);
  const [pageError, setPageError] = useState(0);
  const [planName, setPlanName] = useState("");
  const [courses, setCourses] = useState([]);
  const [warning, setWarning] = useState("");
  const [edit, setEdit] = useState(0);
  const {planId} = useParams();

  const style = css`
    & {
      display: grid;
      width: 100vw;
      height: 100vh;
      max-height: 100vh;
      position: absolute;
      top: 0;
      left: 0;
      right: 0;
      bottom: 0;
      overflow: hidden;
      grid-template-columns: 1rem 7fr 5rem 3fr 1rem;
      grid-template-rows: 50px 50px 1fr auto 0px;
      grid-column-gap: 0px;
      grid-row-gap: 1rem;
      grid-template-areas: 'navbar  navbar  navbar  navbar  navbar'
                           'left    plan    center  search  right'
                           'left    plan    center  search  right'
                           'left    plan    center  search  right'
                           'bottom  bottom  bottom  bottom  bottom';
    }
    
    #navbar {
      grid-area: navbar;
    }
    
    #search {
      grid-area: search;
    }
  `;

  useEffect(() => {
    async function fetchPlan(planId) {
      setLoading(true);
      try {
        const url = `/api/plan/${planId}`;
        let obj = [];

        const response = await fetch(url);
        if (response.ok) {
        // get data from the response
          obj = await response.json();
          // get the courses array from the plan object
          setCourses(obj.courses);
          setPlanName(obj.planName);
          setLoading(false);
        } else {
          // we got a bad status code
          if (response.status === 500) {
            setPageError(500);
          } else {
            setPageError(404);
          }
        }
      } catch (err) {
        // this is a server error
        setPageError(500);
      }
    }

    // only fetch a plan if we are on the edit plan page
    if (planId) {
      setEdit(parseInt(planId));
      fetchPlan(planId);
    }

  }, [planId]);

  // adds a new course to the courses array
  function handleAddCourse(course) {
    // check that new course isn't already in array
    for (let i = 0; i < courses.length; i++) {
      // check for duplicate courses
      if (course.courseId === courses[i].courseId) {
        setWarning("This course is already in your plan.");
        return;
      }
    }
    // check for required courses
    if (course.restriction === 1) {
      setWarning("Required courses are not allowed.");
      return;
    }
    // check for graduate courses
    if (course.restriction === 2) {
      setWarning("Graduate courses are not allowed.");
      return;
    }
    // see if the course has a defined number of credits
    if (isNaN(course.credits)) {

      // check with the user to see how many credits should be applied
      const creditArray = course.credits.split(" to ");
      if (creditArray.length >= 2) {

        // prompt the user to enter the credits
        let credits = prompt(
          `Please select between ${creditArray[0]} and ${creditArray[1]} credits`, ""
        );
        credits = parseInt(credits, 10);

        // see if the user entered a valid number of credits
        if (credits >= creditArray[0] && credits <= creditArray[1] && !isNaN(credits)) {
          course.credits = credits;
          setCourses(prev => [...prev, course]);
          setWarning("");
        } else {
          setWarning("Invalid credit hours selected for course.");
        }

      }

    } else {
      // add the new course
      setCourses(prev => [...prev, course]);
      setWarning("");
    }
  }

  // removes a course from the courses array
  function handleRemoveCourse(course) {
    setCourses(courses.filter(prev => prev.courseId !== course.courseId));
  }

  if (!pageError) {
    return (
      <div className="student-create-plan" css={style}>
        <PageSpinner loading={loading} subloading={subloading}/>
        <Navbar showSearch={false} searchContent={null}/>
        <EditPlan courses={courses} edit={edit} planName={planName} onLoading={load => setSubloading(load)}
          onChangePlanName={e => setPlanName(e)} onRemoveCourse={e => handleRemoveCourse(e)}  />
        <CourseSearch warning={warning} onAddCourse={e => handleAddCourse(e)}
          onNewWarning={e => setWarning(e)}/>
      </div>
    );
  } else if (pageError === 404) {
    return <PageNotFound />;
  } else {
    return <PageInternalError />;
  }
}