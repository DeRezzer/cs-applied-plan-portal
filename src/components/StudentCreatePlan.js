import React from 'react'
import EditPlan from './EditPlan'
import CourseContainer from './CourseContainer'

export default class StudentCreatePlan extends React.Component {

    render(){
        return(
        <div className="student-create-plan">
            <EditPlan />
            <CourseContainer />
        </div>
        );
    }
}