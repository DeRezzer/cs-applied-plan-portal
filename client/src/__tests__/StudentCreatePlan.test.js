/* eslint-disable no-undef */
import React from "react";
import Enzyme, {mount} from "enzyme";
import StudentCreatePlan from "../components/create_plan/StudentCreatePlan";
import {BrowserRouter as Router} from "react-router-dom";
import Adapter from "enzyme-adapter-react-16";
import {MemoryRouter} from "react-router-dom";
import setCookies from "../cookieInfo";
import Cookies from "js-cookie";

Enzyme.configure({adapter: new Adapter()});

describe("Student create plan", () => {

  beforeEach(() => {
    // create a mock function using jest.fn()
    const mockSet = jest.fn();

    // here we are trying to mock the 'set' functionality of Cookie
    Cookies.set = mockSet;

    // call the set method of Cookies
    setCookies("userId", "50734529811");
    setCookies("role", "0");

  });

  it("renders successfully", () => {
    mount(
      <MemoryRouter initialEntries={["/createPlan"]}>
        <StudentCreatePlan />
      </MemoryRouter>);
  });
});