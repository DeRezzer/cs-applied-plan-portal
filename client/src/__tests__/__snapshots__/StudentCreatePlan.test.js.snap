// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student create plan renders successfully 1`] = `
<div
  className="student-create-plan css-mhsa1c-StudentCreatePlan"
>
  <div
    className="navbar-parent css-12g91iz-Navbar"
    id="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <p
        className="osu-logo"
      >
        OSU CS Applied Plan Portal
      </p>
    </a>
    <div
      className="right-container"
    >
      <div
        className="notification-dropdown css-3i4jcj-Notifications"
      >
        <button
          className="drop-button-notification"
          data-count={0}
        >
          Notifications
          <span
            className="badge"
          />
          <i
            className="fa fa-caret-down"
          />
        </button>
        <div
          className="dropdown-content"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <p>
              No new notifications
            </p>
          </a>
        </div>
      </div>
      <button
        className="logout-button css-1k1xu2y-Logout"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </div>
  <div
    className="css-qwwcyr-EditPlan"
    id="edit-container"
  >
    <div
      id="title"
    >
      <input
        id="title-input"
        onChange={[Function]}
        placeholder="Enter plan name"
        type="text"
        value=""
      />
      <div
        id="submit-plan-error"
      >
        <div
          className="hide css-fzm5ub-ErrorMessage"
        >
          <div
            className="error-message css-fzm5ub-ErrorMessage"
          >
            <p>
              
            </p>
          </div>
        </div>
      </div>
      <div
        id="title-credits"
      >
        <div
          id="credits-count"
        >
          0
        </div>
        <div
          id="credits-label"
        >
          credits
        </div>
      </div>
    </div>
    <div
      className="no-courses-msg"
    >
      <h2
        className="empty-plan-title"
      >
        No courses!
      </h2>
      <h4
        className="empty-plan-description"
      >
        Use the search bar to find courses and add them to your plan.
      </h4>
    </div>
  </div>
  <div
    className="css-g91w1x-CourseContainer"
    id="search"
  >
    <div
      className="search-title"
    >
      Search
    </div>
    <div
      className="search-container"
    >
      <form
        className="form my-2 my-lg-0"
        onSubmit={[Function]}
      >
        <input
          className="form-control mr-sm-2"
          id="search-container"
          name="search"
          placeholder="Search for courses..."
          type="text"
        />
      </form>
      <button
        className="search-button"
        onClick={[Function]}
        type="submit"
      >
        Search
      </button>
    </div>
    <form
      className="course-filter form-group"
    >
      <div>
        <select
          onChange={[Function]}
          value="*"
        >
          <option
            className="form-control"
            value="*"
          >
            Any Subject
          </option>
          <option
            className="form-control"
            value="ALS"
          >
            ALS
             - 
            Academic Learning Services
          </option>
          <option
            className="form-control"
            value="ACTG"
          >
            ACTG
             - 
            Accounting
          </option>
          <option
            className="form-control"
            value="AHE"
          >
            AHE
             - 
            Adult Ed & Higher Ed Leadershp
          </option>
          <option
            className="form-control"
            value="AAE"
          >
            AAE
             - 
            Aeronautical & Astronaut. Eng.
          </option>
          <option
            className="form-control"
            value="AS"
          >
            AS
             - 
            Aerospace Studies
          </option>
          <option
            className="form-control"
            value="AED"
          >
            AED
             - 
            Agricultural Education
          </option>
          <option
            className="form-control"
            value="AGRI"
          >
            AGRI
             - 
            Agricultural Sci
          </option>
          <option
            className="form-control"
            value="AG"
          >
            AG
             - 
            Agriculture-General
          </option>
          <option
            className="form-control"
            value="ASL"
          >
            ASL
             - 
            American Sign Language
          </option>
          <option
            className="form-control"
            value="AMS"
          >
            AMS
             - 
            American Studies Program
          </option>
          <option
            className="form-control"
            value="ANS"
          >
            ANS
             - 
            Animal Sciences
          </option>
          <option
            className="form-control"
            value="ANTH"
          >
            ANTH
             - 
            Anthropology
          </option>
          <option
            className="form-control"
            value="AEC"
          >
            AEC
             - 
            Applied Economics
          </option>
          <option
            className="form-control"
            value="AJ"
          >
            AJ
             - 
            Applied Journalism
          </option>
          <option
            className="form-control"
            value="ARAB"
          >
            ARAB
             - 
            Arabic
          </option>
          <option
            className="form-control"
            value="ARE"
          >
            ARE
             - 
            Architectural Engineering
          </option>
          <option
            className="form-control"
            value="ART"
          >
            ART
             - 
            Art
          </option>
          <option
            className="form-control"
            value="ASN"
          >
            ASN
             - 
            Asian Languages and Culture
          </option>
          <option
            className="form-control"
            value="ATS"
          >
            ATS
             - 
            Atmospheric Sciences
          </option>
          <option
            className="form-control"
            value="BB"
          >
            BB
             - 
            Biochemistry and Biophysics
          </option>
          <option
            className="form-control"
            value="BHS"
          >
            BHS
             - 
            BioHealth Sciences
          </option>
          <option
            className="form-control"
            value="BEE"
          >
            BEE
             - 
            Biological & Ecological Engr
          </option>
          <option
            className="form-control"
            value="BDS"
          >
            BDS
             - 
            Biological Data Sciences
          </option>
          <option
            className="form-control"
            value="BIOE"
          >
            BIOE
             - 
            Biological Engineering
          </option>
          <option
            className="form-control"
            value="BI"
          >
            BI
             - 
            Biology
          </option>
          <option
            className="form-control"
            value="BRR"
          >
            BRR
             - 
            Bioresource Research
          </option>
          <option
            className="form-control"
            value="BOT"
          >
            BOT
             - 
            Botany and Plant Pathology
          </option>
          <option
            className="form-control"
            value="BA"
          >
            BA
             - 
            Business Administration
          </option>
          <option
            className="form-control"
            value="CBEE"
          >
            CBEE
             - 
            Chem, Bio, Enviro Engineering
          </option>
          <option
            className="form-control"
            value="CHE"
          >
            CHE
             - 
            Chemical Engineering
          </option>
          <option
            className="form-control"
            value="CH"
          >
            CH
             - 
            Chemistry
          </option>
          <option
            className="form-control"
            value="CHN"
          >
            CHN
             - 
            Chinese
          </option>
          <option
            className="form-control"
            value="CCE"
          >
            CCE
             - 
            Civil and Construction Engr
          </option>
          <option
            className="form-control"
            value="CE"
          >
            CE
             - 
            Civil Engineering
          </option>
          <option
            className="form-control"
            value="CSSA"
          >
            CSSA
             - 
            College Student Services Admin
          </option>
          <option
            className="form-control"
            value="COMM"
          >
            COMM
             - 
            Communication
          </option>
          <option
            className="form-control"
            value="CS"
          >
            CS
             - 
            Computer Science
          </option>
          <option
            className="form-control"
            value="CEM"
          >
            CEM
             - 
            Construction Engineering Mngmt
          </option>
          <option
            className="form-control"
            value="COUN"
          >
            COUN
             - 
            Counseling
          </option>
          <option
            className="form-control"
            value="CSS"
          >
            CSS
             - 
            Crop & Soil Science
          </option>
          <option
            className="form-control"
            value="CROP"
          >
            CROP
             - 
            Crop Science
          </option>
          <option
            className="form-control"
            value="DSGN"
          >
            DSGN
             - 
            Design
          </option>
          <option
            className="form-control"
            value="DHE"
          >
            DHE
             - 
            Design and Human Environment
          </option>
          <option
            className="form-control"
            value="ECON"
          >
            ECON
             - 
            Economics
          </option>
          <option
            className="form-control"
            value="ED"
          >
            ED
             - 
            Education
          </option>
          <option
            className="form-control"
            value="ECE"
          >
            ECE
             - 
            Electrical & Computer Engineer
          </option>
          <option
            className="form-control"
            value="ESE"
          >
            ESE
             - 
            Energy Systems Engineering
          </option>
          <option
            className="form-control"
            value="ENGR"
          >
            ENGR
             - 
            Engineering Science
          </option>
          <option
            className="form-control"
            value="ENG"
          >
            ENG
             - 
            English
          </option>
          <option
            className="form-control"
            value="ENT"
          >
            ENT
             - 
            Entomology
          </option>
          <option
            className="form-control"
            value="EAH"
          >
            EAH
             - 
            Environmental Arts & Humanities
          </option>
          <option
            className="form-control"
            value="ENVE"
          >
            ENVE
             - 
            Environmental Engineering
          </option>
          <option
            className="form-control"
            value="ENSC"
          >
            ENSC
             - 
            Environmental Sciences
          </option>
          <option
            className="form-control"
            value="ES"
          >
            ES
             - 
            Ethnic Studies
          </option>
          <option
            className="form-control"
            value="FILM"
          >
            FILM
             - 
            Film Studies
          </option>
          <option
            className="form-control"
            value="FIN"
          >
            FIN
             - 
            Finance
          </option>
          <option
            className="form-control"
            value="FW"
          >
            FW
             - 
            Fisheries and Wildlife
          </option>
          <option
            className="form-control"
            value="FCSJ"
          >
            FCSJ
             - 
            Food in Culture, Social Justice
          </option>
          <option
            className="form-control"
            value="FST"
          >
            FST
             - 
            Food Science and Technology
          </option>
          <option
            className="form-control"
            value="FES"
          >
            FES
             - 
            Forest Ecosystems and Society
          </option>
          <option
            className="form-control"
            value="FE"
          >
            FE
             - 
            Forest Engineering
          </option>
          <option
            className="form-control"
            value="FOR"
          >
            FOR
             - 
            Forestry
          </option>
          <option
            className="form-control"
            value="FR"
          >
            FR
             - 
            French
          </option>
          <option
            className="form-control"
            value="GS"
          >
            GS
             - 
            General Science
          </option>
          <option
            className="form-control"
            value="GEOG"
          >
            GEOG
             - 
            Geography
          </option>
          <option
            className="form-control"
            value="GPH"
          >
            GPH
             - 
            Geophysics
          </option>
          <option
            className="form-control"
            value="GEO"
          >
            GEO
             - 
            Geosciences
          </option>
          <option
            className="form-control"
            value="GER"
          >
            GER
             - 
            German
          </option>
          <option
            className="form-control"
            value="GRAD"
          >
            GRAD
             - 
            Graduate Education
          </option>
          <option
            className="form-control"
            value="GD"
          >
            GD
             - 
            Graphic Design
          </option>
          <option
            className="form-control"
            value="HHS"
          >
            HHS
             - 
            Health and Human Sciences
          </option>
          <option
            className="form-control"
            value="HEBR"
          >
            HEBR
             - 
            Hebrew
          </option>
          <option
            className="form-control"
            value="HST"
          >
            HST
             - 
            History
          </option>
          <option
            className="form-control"
            value="HSTS"
          >
            HSTS
             - 
            History of Science
          </option>
          <option
            className="form-control"
            value="HC"
          >
            HC
             - 
            Honors College
          </option>
          <option
            className="form-control"
            value="HORT"
          >
            HORT
             - 
            Horticulture
          </option>
          <option
            className="form-control"
            value="HM"
          >
            HM
             - 
            Hospitality Management
          </option>
          <option
            className="form-control"
            value="HDFS"
          >
            HDFS
             - 
            Human Dev and Family Sciences
          </option>
          <option
            className="form-control"
            value="HEST"
          >
            HEST
             - 
            Humanitarian Engr Sci & Tech
          </option>
          <option
            className="form-control"
            value="IE"
          >
            IE
             - 
            Industrial and Mfg Engineering
          </option>
          <option
            className="form-control"
            value="IB"
          >
            IB
             - 
            Integrative Biology
          </option>
          <option
            className="form-control"
            value="IEPA"
          >
            IEPA
             - 
            Intensive English Pgm Acad Eng
          </option>
          <option
            className="form-control"
            value="IEPG"
          >
            IEPG
             - 
            Intensive English Pgm Gen Engl
          </option>
          <option
            className="form-control"
            value="IEPH"
          >
            IEPH
             - 
            Intensive English Pgm Gen Engl
          </option>
          <option
            className="form-control"
            value="IST"
          >
            IST
             - 
            Interdisciplinary Programs
          </option>
          <option
            className="form-control"
            value="INTL"
          >
            INTL
             - 
            International Degree
          </option>
          <option
            className="form-control"
            value="IT"
          >
            IT
             - 
            Italian
          </option>
          <option
            className="form-control"
            value="JPN"
          >
            JPN
             - 
            Japanese
          </option>
          <option
            className="form-control"
            value="KIN"
          >
            KIN
             - 
            Kinesiology
          </option>
          <option
            className="form-control"
            value="KOR"
          >
            KOR
             - 
            Korean
          </option>
          <option
            className="form-control"
            value="LEAD"
          >
            LEAD
             - 
            Leadership
          </option>
          <option
            className="form-control"
            value="LA"
          >
            LA
             - 
            Liberal Arts
          </option>
          <option
            className="form-control"
            value="LS"
          >
            LS
             - 
            Liberal Studies
          </option>
          <option
            className="form-control"
            value="LIB"
          >
            LIB
             - 
            Library & Information Science
          </option>
          <option
            className="form-control"
            value="LING"
          >
            LING
             - 
            Linguistics
          </option>
          <option
            className="form-control"
            value="MGMT"
          >
            MGMT
             - 
            Management
          </option>
          <option
            className="form-control"
            value="MFGE"
          >
            MFGE
             - 
            Manufacturing Engineering
          </option>
          <option
            className="form-control"
            value="MRM"
          >
            MRM
             - 
            Marine Resource Management
          </option>
          <option
            className="form-control"
            value="MRKT"
          >
            MRKT
             - 
            Marketing
          </option>
          <option
            className="form-control"
            value="MNR"
          >
            MNR
             - 
            Master of Natural Resources
          </option>
          <option
            className="form-control"
            value="MPP"
          >
            MPP
             - 
            Master of Public Policy
          </option>
          <option
            className="form-control"
            value="MATS"
          >
            MATS
             - 
            Materials Science
          </option>
          <option
            className="form-control"
            value="MTH"
          >
            MTH
             - 
            Mathematics
          </option>
          <option
            className="form-control"
            value="MIME"
          >
            MIME
             - 
            Mech/Ind/Mfg Engineering
          </option>
          <option
            className="form-control"
            value="ME"
          >
            ME
             - 
            Mechanical Engineering
          </option>
          <option
            className="form-control"
            value="MB"
          >
            MB
             - 
            Microbiology
          </option>
          <option
            className="form-control"
            value="MS"
          >
            MS
             - 
            Military Science
          </option>
          <option
            className="form-control"
            value="MCB"
          >
            MCB
             - 
            Molecular & Cellular Biology
          </option>
          <option
            className="form-control"
            value="MUS"
          >
            MUS
             - 
            Music
          </option>
          <option
            className="form-control"
            value="MUP"
          >
            MUP
             - 
            Music (Studio)
          </option>
          <option
            className="form-control"
            value="MUED"
          >
            MUED
             - 
            Music Education
          </option>
          <option
            className="form-control"
            value="NR"
          >
            NR
             - 
            Natural Resources
          </option>
          <option
            className="form-control"
            value="NS"
          >
            NS
             - 
            Naval Science
          </option>
          <option
            className="form-control"
            value="NMC"
          >
            NMC
             - 
            New Media Communications
          </option>
          <option
            className="form-control"
            value="NSE"
          >
            NSE
             - 
            Nuclear Science & Engineering
          </option>
          <option
            className="form-control"
            value="NUR"
          >
            NUR
             - 
            Nursing
          </option>
          <option
            className="form-control"
            value="NUTR"
          >
            NUTR
             - 
            Nutrition
          </option>
          <option
            className="form-control"
            value="OEAS"
          >
            OEAS
             - 
            Ocean Earth & Atmospheric Sci
          </option>
          <option
            className="form-control"
            value="OC"
          >
            OC
             - 
            Oceanography
          </option>
          <option
            className="form-control"
            value="OP"
          >
            OP
             - 
            Outdoor Products
          </option>
          <option
            className="form-control"
            value="PAX"
          >
            PAX
             - 
            Peace Studies
          </option>
          <option
            className="form-control"
            value="PHAR"
          >
            PHAR
             - 
            Pharmacy
          </option>
          <option
            className="form-control"
            value="PHL"
          >
            PHL
             - 
            Philosophy
          </option>
          <option
            className="form-control"
            value="PAC"
          >
            PAC
             - 
            Physical Activity Courses
          </option>
          <option
            className="form-control"
            value="PT"
          >
            PT
             - 
            Physical Therapy/Therapist
          </option>
          <option
            className="form-control"
            value="PH"
          >
            PH
             - 
            Physics
          </option>
          <option
            className="form-control"
            value="PBG"
          >
            PBG
             - 
            Plant Breeding & Genetics
          </option>
          <option
            className="form-control"
            value="PS"
          >
            PS
             - 
            Political Science
          </option>
          <option
            className="form-control"
            value="PSM"
          >
            PSM
             - 
            Professional Science Masters
          </option>
          <option
            className="form-control"
            value="PSY"
          >
            PSY
             - 
            Psychology
          </option>
          <option
            className="form-control"
            value="H"
          >
            H
             - 
            Public Health
          </option>
          <option
            className="form-control"
            value="PPOL"
          >
            PPOL
             - 
            Public Policy
          </option>
          <option
            className="form-control"
            value="QS"
          >
            QS
             - 
            Queer Studies
          </option>
          <option
            className="form-control"
            value="RNG"
          >
            RNG
             - 
            Rangeland Ecology & Management
          </option>
          <option
            className="form-control"
            value="REL"
          >
            REL
             - 
            Religious Studies
          </option>
          <option
            className="form-control"
            value="ROB"
          >
            ROB
             - 
            Robotics
          </option>
          <option
            className="form-control"
            value="RS"
          >
            RS
             - 
            Rural Studies
          </option>
          <option
            className="form-control"
            value="RUS"
          >
            RUS
             - 
            Russian
          </option>
          <option
            className="form-control"
            value="SED"
          >
            SED
             - 
            Science & Mathematics Educ
          </option>
          <option
            className="form-control"
            value="SSCI"
          >
            SSCI
             - 
            Social Science
          </option>
          <option
            className="form-control"
            value="SOC"
          >
            SOC
             - 
            Sociology
          </option>
          <option
            className="form-control"
            value="SE"
          >
            SE
             - 
            Software Engineering
          </option>
          <option
            className="form-control"
            value="SOIL"
          >
            SOIL
             - 
            Soil Science
          </option>
          <option
            className="form-control"
            value="SPAN"
          >
            SPAN
             - 
            Spanish
          </option>
          <option
            className="form-control"
            value="ST"
          >
            ST
             - 
            Statistics
          </option>
          <option
            className="form-control"
            value="SUS"
          >
            SUS
             - 
            Sustainability
          </option>
          <option
            className="form-control"
            value="SNR"
          >
            SNR
             - 
            Sustainable Natural Resources
          </option>
          <option
            className="form-control"
            value="TA"
          >
            TA
             - 
            Theatre Arts
          </option>
          <option
            className="form-control"
            value="TRAL"
          >
            TRAL
             - 
            Tourism, Recreat, Adven. Lead.
          </option>
          <option
            className="form-control"
            value="TOX"
          >
            TOX
             - 
            Toxicology
          </option>
          <option
            className="form-control"
            value="TCS"
          >
            TCS
             - 
            Twentieth Century Studies
          </option>
          <option
            className="form-control"
            value="UEXP"
          >
            UEXP
             - 
            University Experience
          </option>
          <option
            className="form-control"
            value="VMB"
          >
            VMB
             - 
            Veterinary Medicine Biomedical
          </option>
          <option
            className="form-control"
            value="VMC"
          >
            VMC
             - 
            Veterinary Medicine Clinical
          </option>
          <option
            className="form-control"
            value="WRE"
          >
            WRE
             - 
            Water Resources Engineering
          </option>
          <option
            className="form-control"
            value="WRP"
          >
            WRP
             - 
            Water Resources Policy and Mgt
          </option>
          <option
            className="form-control"
            value="WRS"
          >
            WRS
             - 
            Water Resources Science
          </option>
          <option
            className="form-control"
            value="WGSS"
          >
            WGSS
             - 
            Women, Gender, and Sexuality
          </option>
          <option
            className="form-control"
            value="WSE"
          >
            WSE
             - 
            Wood Science and Engineering
          </option>
          <option
            className="form-control"
            value="WLC"
          >
            WLC
             - 
            World Languages and Cultures
          </option>
          <option
            className="form-control"
            value="WR"
          >
            WR
             - 
            Written English
          </option>
          <option
            className="form-control"
            value="Z"
          >
            Z
             - 
            Zoology
          </option>
        </select>
      </div>
    </form>
    <div
      className="hide css-fzm5ub-ErrorMessage"
    >
      <div
        className="error-message css-fzm5ub-ErrorMessage"
      >
        <p>
          
        </p>
      </div>
    </div>
    <div
      className="search-results"
    >
      <div>
        Search for courses...
      </div>
    </div>
  </div>
</div>
`;
